study.md

```diff
@@ -21,18 +21,18 @@ There is no requirement to follow any links from these pages.
 After doing all the required readings please tell us anything you were confused about.
 
 ```md
-<!-- your answer here -->
+Hoisted functions still confuse me, I'm very tired I will revisit the concept with a fresh mind tomorrow.
```

Comment:
Here's a snippet of code that I wrote in hopes that it'll help you all understand hoisting.  Let me know if it's unclear.
```javascript
// This will output, "[Function: sayHello]".
console.log(sayHello);

// This will work and output, "Hello".
sayHello();

// This will output, "undefined" since the declaration of "sayHi" has been
// hoisted to the top, but its definition has not been.
console.log(sayHi);

// This will throw an error for the reason above.  Since sayHi is undefined,
// it's trying to execute "undefined" as if it were a function.
sayHi();

// This gets hoisted to the top.
function sayHello() {
  console.log('Hello.');
}

// "var sayHi" gets hoisted to the top, but its assignment to function() { ... }
// does not.  Variable declarations using "const" and "let" don't get hoisted.
var sayHi = function() {
  console.log('Hi.');
};

// This will output, "[Function: sayHi]", since it's after the definition of the
// function.
console.log(sayHi);

// This will output, "Hi".
sayHi();
```